### PR TITLE
build: update frozen toolchain to Fedora 40 with clang 18.1.6

### PR DESCRIPTION
**Pull request by avikivity**
This refreshes our dependencies to a supported distribution.

No backport required as this doesn't fix a bug.

### CODE REVIEW

**Comment by avikivity** [comment]
> #### 🔴 CI State: FAILURE
> ❌ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9563/)
> 
> #### Build Details:
> * Duration: 11 min
> * Builder: spider4.cloudius-systems.com

https://github.com/scylladb/scylla-tools-java/pull/395

**Comment by avikivity** [comment]
v2: rebased on next for 56ec839f41d137a39fe0a54ca760200674697c83

File: tools/toolchain/optimized_clang.sh [hunk 1]
```diff
@@ -42,8 +42,8 @@ SCYLLA_BUILD_DIR_FULLPATH="${SCYLLA_DIR}"/"${SCYLLA_BUILD_DIR}"
 SCYLLA_NINJA_FILE_FULLPATH="${SCYLLA_DIR}"/"${SCYLLA_NINJA_FILE}"
 
 # Which LLVM release to build in order to compile Scylla
-LLVM_CLANG_TAG=16.0.6
-CLANG_SUFFIX=16
+LLVM_CLANG_TAG=18.1.6
```
[the remark this answers is not in the document]
Comment (avikivity) [reply tools/toolchain/optimized_clang.sh:45]: yes but I want to be in sync with Fedora

**Comment by tchaikov** [comment]
```
21:14:55  [325/3249] cd tools/java && ./reloc/build_reloc.sh --version $(<../../build/SCYLLA-PRODUCT-FILE)-$(sed 's/-/~/' <../../build/SCYLLA-VERSION-FILE)-$(<../../build/SCYLLA-RELEASE-FILE) --nodeps 
21:14:55  FAILED: tools/java/build/scylla-tools-6.1.0~dev-0.20240610.7172df3967a3.noarch.tar.gz 
21:14:55  cd tools/java && ./reloc/build_reloc.sh --version $(<../../build/SCYLLA-PRODUCT-FILE)-$(sed 's/-/~/' <../../build/SCYLLA-VERSION-FILE)-$(<../../build/SCYLLA-RELEASE-FILE) --nodeps 
...
21:14:55      [unzip] Expanding: /jenkins/workspace/scylla-master/scylla-ci/scylla/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946.jar into /jenkins/workspace/scylla-master/scylla-ci/scylla/tools/java/build/lib/jars
21:14:55  
21:14:55  check-gen-cql3-grammar:
21:14:55  
21:14:55  gen-cql3-grammar:
21:14:55       [echo] Building Grammar /jenkins/workspace/scylla-master/scylla-ci/scylla/tools/java/src/antlr/Cql.g  ...
21:14:55  Traceback (most recent call last):
21:14:55    File "/jenkins/workspace/scylla-master/scylla-ci/scylla/tools/java/scripts/create-relocatable-package.py", line 67, in <module>
21:14:55      ar.reloc_add('build/apache-cassandra-{}.jar'.format(version), arcname='lib/apache-cassandra-{}.jar'.format(version))
21:14:55    File "/jenkins/workspace/scylla-master/scylla-ci/scylla/tools/java/scripts/create-relocatable-package.py", line 33, in reloc_add
21:14:55      return self.add(name, arcname="{}/{}".format(RELOC_PREFIX, arcname))
21:14:55             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
21:14:55    File "/usr/lib64/python3.12/tarfile.py", line 2164, in add
21:14:55      tarinfo = self.gettarinfo(name, arcname)
21:14:55                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
21:14:55    File "/usr/lib64/python3.12/tarfile.py", line 2037, in gettarinfo
21:14:55      statres = os.lstat(name)
21:14:55                ^^^^^^^^^^^^^^
21:14:55  FileNotFoundError: [Errno 2] No such file or directory: 'build/apache-cassandra-6.1.0~dev-0.20240610.7172df3967a3.jar'
```

**Comment by avikivity** [comment]
Looks like a bug in ant.

**Comment by avikivity** [comment]
https://github.com/scylladb/scylla-tools-java/pull/396

**Comment by avikivity** [comment]
Queued a fix just to test it out.

**Comment by avikivity** [comment]
@yaronkaikov I don't undertstand why "Container Test" failed (or what it tests)

**Comment by yaronkaikov** [comment]
> @yaronkaikov I don't undertstand why "Container Test" failed (or what it tests)

Because of
```
[2024-06-10T17:37:33.350Z] + podman exec -it 415b43dceb88838d165589cd89d91ae37db77fc2f2d612bfdf115e2b400d419e nodetool status

[2024-06-10T17:37:33.350Z] Error: can only create exec sessions on running containers: container state improper
```

We try 5 times (with some wait time in between) and it didn't manage to get proper status

File: tools/toolchain/optimized_clang.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (tchaikov) [reply tools/toolchain/optimized_clang.sh:45]: ack.

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9581/) ❌ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9581/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9581/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10011/) ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/1986/)
> #### Failed Tests (1342/30692):
> 
>     * [test_sstableloder_scenario[without_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest___test_sstableloder_scenario_without_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bwithout_flag%5D)
> 
>     * [test_sstableloder_scenario[v_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest___test_sstableloder_scenario_v_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bv_flag%5D)
> 
>     * [test_sstableloder_scenario[nb_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest___test_sstableloder_scenario_nb_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bnb_flag%5D)
> 
>     * [test_sstableloder_scenario[without_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest_with_topology_changes___test_sstableloder_scenario_without_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bwithout_flag%5D)
> 
>     * [test_sstableloder_scenario[v_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest_with_topology_changes___test_sstableloder_scenario_v_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bv_flag%5D)
> 
>     * [test_sstableloder_scenario[nb_flag]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest_with_topology_changes___test_sstableloder_scenario_nb_flag_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_sstableloder_scenario%5Bnb_flag%5D)
> 
>     * [test_manual_bootstrap](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/bootstrap_test/TestBootstrap/Tests___dtest___test_manual_bootstrap) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_manual_bootstrap)
> 
>     * [test_local_quorum_bootstrap](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/bootstrap_test/TestBootstrap/Tests___dtest___test_local_quorum_bootstrap) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_local_quorum_bootstrap)
> 
>     * [test_manual_bootstrap](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/bootstrap_test/TestBootstrap/Tests___dtest_with_topology_changes___test_manual_bootstrap) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_manual_bootstrap)
> 
>     * [test_local_quorum_bootstrap](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9581/testReport/junit/bootstrap_test/TestBootstrap/Tests___dtest_with_topology_changes___test_local_quorum_bootstrap) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_local_quorum_bootstrap)
> 
> 
> #### Build Details:
> 
>     * Duration: 3 hr 0 min
> 
>     * Builder: spider5.cloudius-systems.com

i think we need https://github.com/scylladb/scylla-tools-java/pull/397 .

**Comment by avikivity** [comment]
> > @yaronkaikov I don't undertstand why "Container Test" failed (or what it tests)
> 
> Because of
> 
> ```
> [2024-06-10T17:37:33.350Z] + podman exec -it 415b43dceb88838d165589cd89d91ae37db77fc2f2d612bfdf115e2b400d419e nodetool status
> 
> [2024-06-10T17:37:33.350Z] Error: can only create exec sessions on running containers: container state improper
> ```
> 
> We try 5 times (with some wait time in between) and it didn't manage to get proper status

It was a true positive.

**Comment by avikivity** [comment]
Fix for the container test failure: https://github.com/scylladb/scylladb/pull/19236

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9607/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9607/) ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2011/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10035/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9607/)
> 
> #### Failed Tests (266/30712):
> * [test_connectivity_with_cqlsh](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cloud_bundle_test/TestScyllaCloudBundle/Tests___dtest_with_topology_changes___test_connectivity_with_cqlsh) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_connectivity_with_cqlsh)
> * [test_connectivity_with_cqlsh](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cloud_bundle_test/TestScyllaCloudBundle/Tests___dtest___test_connectivity_with_cqlsh) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_connectivity_with_cqlsh)
> * [test_cast_udt](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_cast_test/TestCQLCast/Tests___dtest_with_topology_changes___test_cast_udt) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_cast_udt)
> * [test_cast_udt](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_cast_test/TestCQLCast/Tests___dtest___test_cast_udt) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_cast_udt)
> * [test_tracing_info_for_all_partitions](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_for_all_partitions) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_for_all_partitions)
> * [test_tracing_info_for_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_for_mv) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_for_mv)
> * [test_tracing_info_selecting_by_one_key](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_selecting_by_one_key) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_selecting_by_one_key)
> * [test_tracing_info_for_index_read_range](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_for_index_read_range) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_for_index_read_range)
> * [test_tracing_info_for_index_read_one](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_for_index_read_one) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_for_index_read_one)
> * [test_tracing_info_read_from_several_sstables](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9607/testReport/junit/cql_tracing_addition_test/TestTracingReadAccess/Tests___dtest_with_topology_changes___test_tracing_info_read_from_several_sstables) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_info_read_from_several_sstables)
> 
> #### Build Details:
> * Duration: 5 hr 27 min
> * Builder: i-022b694ee88bad3a2 (m5ad.8xlarge)

i sampled 10 failures, all them were caused by the warning message addressed by https://github.com/scylladb/python-driver/commit/82b4863a58a59a28d27848c6eec22f503be665e0, which has not been included any scylla-driver release. but neither does the unified package include the latest scylla-driver, which is 3.26.8, see https://pypi.org/project/scylla-driver/ .

the error message indicates that the used scylla-driver is 3.26.5. so we also need to update https://github.com/scylladb/scylla-cqlsh/blob/0d58e5ceb4216d131fb07803f904b99e8d11e805/requirements.txt#L1 to use the new release.

```
E         + /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/19205/share/cassandra/libexec/../lib/scylla-driver-3.26.5.zip/cassandra/cluster.py:5419: SyntaxWarning: invalid escape sequence '\s'
```
and
```
16:09:31,207 699     ccm                            DEBUG    cluster.py          :754  | test_repair_while_table_is_dropped: node1: (EE) /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/19205/share/cassandra/libexec/../lib/scylla-driver-3.26.5.zip/cassandra/cluster.py:5419: SyntaxWarning: invalid escape sequence '\s'
16:09:31,207 699     ccm                            DEBUG    cluster.py          :754  | test_repair_while_table_is_dropped: node1: (EE) /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/19205/share/cassandra/libexec/../lib/scylla-driver-3.26.5.zip/cassandra/cluster.py:5419: SyntaxWarning: invalid escape sequence '\s'
16:09:31,207 699     ccm                            DEBUG    cluster.py          :754  | test_repair_while_table_is_dropped: node1: (EE) <stdin>:2:Table 'cf_del0' not found in keyspace 'ks'
```

@fruch hi Israel, is it possible to create a new release which includes https://github.com/scylladb/python-driver/commit/82b4863a58a59a28d27848c6eec22f503be665e0, so that we don't print this error message with python 3.12 ?

**Comment by avikivity** [comment]
@fruch please update

**Comment by avikivity** [comment]
@fruch what's the status?

**Comment by fruch** [comment]
> @fruch what's the status?

didn't saw this one until now, I
don't know when I'll get to it.
maybe @Lorak-mmk can do that

@roydahan FYI

**Comment by Lorak-mmk** [comment]
I can release new python driver and make a PR to cqlsh to update the requirement. If there is a need to release cqlsh too, then @fruch would need to explain to me how to do it again (he did it once, but it was a long time ago).

**Comment by Lorak-mmk** [comment]
Published driver 3.26.9
PR to cqlsh is here: https://github.com/scylladb/scylla-cqlsh/pull/92

**Comment by Lorak-mmk** [comment]
Released cqlsh 6.0.21: https://pypi.org/project/scylla-cqlsh/6.0.21/ @avikivity

**Comment by avikivity** [comment]
Rebased past 3cfb0503a989c561ae5347a653856abb94af428a and regenerated the image.

**Comment by fruch** [comment]
Seems like there are still deprecation warnings that break a few of the tests.

I think tests could be fixed to ignore, until it would be addressed in cqlsh

The one concerning is the cloud bundle that fails, it's not clear why it fails to connect, it should be raised with cqlsh

**Comment by tchaikov** [comment]
> Seems like there are still deprecation warnings that break a few of the tests.
> 
> I think tests could be fixed to ignore, until it would be addressed in cqlsh
> 
> The one concerning is the cloud bundle that fails, it's not clear why it fails to connect, it should be raised with cqlsh

some of them should be fixed by https://github.com/scylladb/python-driver/pull/337. tested with
``console
$ JAVA_HOME=/usr/lib/jvm/jre-11 pytest -s --log-cli-level=debug -c $PWD/pytest.ini --cassandra-dir $HOME/dev/scylladb/build/RelWithDebInfo cql_tracing_addition_test.py::TestTracingReadAccess
$ JAVA_HOME=/usr/lib/jvm/jre-11 pytest -s --log-cli-level=debug -c $PWD/pytest.ini --cassandra-dir $HOME/dev/scylladb/build/RelWithDebInfo cql_tracing_addition_test.py::TestTracingReadAccess
```

but 
```console
$ JAVA_HOME=/usr/lib/jvm/jre-11 pytest -s --log-cli-level=debug -c $PWD/pytest.ini --cassandra-dir $HOME/dev/scylladb/build/RelWithDebInfo  cloud_bundle_test.py::TestScyllaCloudBundle::test_connectivity_with_cqlsh
...
16:11:22,616 ccm             DEBUG cluster.py     :754  | node1: (EE) Connection error: ('Unable to connect to any servers', {'127.0.30.1:443:cql.cluster-id.scylla.com': ConnectionRefusedError(111, "Tried connecting to [('127.0.30.1', 443)]. Last error: Connection refused")})
16:11:22,616 ccm             DEBUG cluster.py     :754  | node1:
```

still fails. log from node1:
```
INFO  2024-06-23 16:11:15,175 [shard 0:comp] compaction - [Compact system_schema.tables 293f7950-3138-11ef-bed1-e8868114d9e4] Compacted 2 sstables to [/home/kefu/.dtest/dtest-_scgi6zm/test/node1/data/system_schema/tables-afddfb9dbc1e30688
056eed6c302ba09/me-3gh8_0mqr_112vl2wgs9pi0h6pyc-big-Data.db:level=0]. 20kB to 14kB (~67% of original) in 2ms = 10MB/s. ~256 total partitions merged to 5.
INFO  2024-06-23 16:11:15,687 [shard 0:stmt] cql_server - exception while processing connection: std::system_error (error GnuTLS:-15, An unexpected TLS packet was received.)
INFO  2024-06-23 16:11:16,043 [shard 0:stmt] cql_server - exception while processing connection: std::system_error (error GnuTLS:-15, An unexpected TLS packet was received.)
INFO  2024-06-23 16:11:18,100 [shard 1:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.cer"}
INFO  2024-06-23 16:11:18,104 [shard 1:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.cer"}
INFO  2024-06-23 16:11:18,104 [shard 0:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.cer"}
INFO  2024-06-23 16:11:18,109 [shard 1:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.pem", "/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.key"}
INFO  2024-06-23 16:11:18,113 [shard 1:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.pem", "/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.key"}
INFO  2024-06-23 16:11:18,113 [shard 0:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.cer"}
INFO  2024-06-23 16:11:18,118 [shard 0:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.pem", "/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.key"}
INFO  2024-06-23 16:11:18,122 [shard 0:stmt] cql_server - Reloaded {"/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.pem", "/home/kefu/.dtest/dtest-_scgi6zm/test/ccm_node.key"}
```

**Comment by fruch** [comment]
@tchaikov 

seems like  https://github.com/scylladb/scylla-cqlsh/issues/90 
i.e. python3.12 is used for cqlsh and it fails back to asyncio, and that isn't work for SSL.

so https://github.com/scylladb/scylladb/pull/19385 probably gonna help.

but it's not yet finished and working.

**Comment by fruch** [comment]
https://github.com/scylladb/scylladb/pull/19385 is getting along, but there are multiple moving part (change in cqlsh, core, scylla-pkg)

in the meanwhile, I think `cloud_bundle_test.TestScyllaCloudBundle.test_connectivity_with_cqlsh` can be marked as with an issue to be skipped, so we can let this one pass.

**Comment by mykaul** [comment]
> #19385 is getting along, but there are multiple moving part (change in cqlsh, core, scylla-pkg)
> 
> in the meanwhile, I think `cloud_bundle_test.TestScyllaCloudBundle.test_connectivity_with_cqlsh` can be marked as with an issue to be skipped, so we can let this one pass.

The bundle isn't used anywhere anymore, certainly can be skipped / dropped.

**Comment by fruch** [comment]
> > #19385 is getting along, but there are multiple moving part (change in cqlsh, core, scylla-pkg)
> > in the meanwhile, I think `cloud_bundle_test.TestScyllaCloudBundle.test_connectivity_with_cqlsh` can be marked as with an issue to be skipped, so we can let this one pass.
> 
> The bundle isn't used anywhere anymore, certainly can be skipped / dropped.

it doesn't point to the fact the our cqlsh might be broken, and fail to work work in some cases with TLS
 i.e. we know for fact that the asyncio eventloop of the driver, has issues with TLS:

see: https://github.com/scylladb/python-driver/pull/327#issuecomment-2151598758

looking the other way around, might come biting (as always)

**Comment by avikivity** [comment]
update: rebased on top of 3c7af28725

**Comment by avikivity** [comment]
@nyh please review

**Comment by avikivity** [comment]
Or maybe @denesb ?

**Review by nyh — APPROVED** [review]

**Pull request closed without merging**